sandbox/lua/decoders/rsyslog.lua

```diff
@@ -93,14 +96,16 @@ function process_message ()
         fields.syslogtag = nil
     end
 
-    msg.Hostname = fields.hostname or fields.source
-    fields.hostname = nil
-    fields.source = nil
+    if not hostname_keep then
+        msg.Hostname = fields.hostname or fields.source
+        fields.hostname = nil
+        fields.source = nil
+    end
 
     msg.Payload = fields.msg
     fields.msg = nil
 
     msg.Fields = fields
-    inject_message(msg)
+    if not pcall(inject_message, msg) then return -1 end
```

Comment:
I assume this was added to catch an output_limit error since the grammar ensures the message is well formed.  So we are changing the behaviour from termination the dropping the data... I am good with that, any objections?

CHANGES.txt should be updated